### ISSUE TITLE
Cloudant - Model property with type number and marked ID expects string as value

### ISSUE THREAD

**Issue by cgole** [strongloop/loopback-connector-cloudant#31]
PMR link: https://jazzop27.rtp.raleigh.ibm.com:9443/ccm/web/projects/CloudOE#action=com.ibm.team.workitem.viewWorkItem&id=210887

## Problem:
Attempting to insert a record into Cloudant with a numeric ID will fail, since the inserted object does not have an id of type String.

## Potential Fix
Investigate whether or not calling `.toString()` on the `_id` parameter before an insert request is made will fix the issue.

## Acceptance Criteria
- Numeric IDs should be valid for inserting records into Cloudant
- Test case(s) should validate this behaviour
- Document this workaround on loopback-connector-cloudant

**Comment by cgole** [comment]
Since cloudant uses documents to store data then we can only use type `string` for id:
see https://docs.cloudant.com/document.html

**Comment by jannyHou** [comment]
Add some details from discussion:

```json
"id": {
     "type": "number",
     "id": true,
     "required": true,
     "generated": false
}
```
Will see error 
> Document id must be a string

I am figuring out how loopback converts the id type to string, will update details later. Not sure which one is better, giving a warning or throwing an error, but we should add some validation/check instead of silently changing it to another type.

**Comment by cgole** [comment]
@deepakrkris this needs to be fixed asap. Customer is pushing for a fix

**Comment by ssh24** [comment]
Issue fixed on #115. Moving it to TBR.
